src/plugins/kibana_utils/common/persistable_state/index.ts

```diff
@@ -8,7 +8,14 @@
 
 import { SavedObjectReference } from '../../../../core/types';
 
-export type SerializableValue = string | number | boolean | null | undefined | SerializableState;
+export type SerializableValue =
+  | string
+  | number
+  | boolean
+  | null
+  | undefined
+  | unknown
+  | SerializableState;
```

Comment:
Added `unknown` to the options for `SerializableValue` in order to work with existing `AlertTypeParams` type, which was `Record<string, unknown>`. Is this ok?

Reply:
unknown can be something that is not serializable, but no strong opinion. what do you think @stacey-gammon ?

Reply:
Yea, I don't think this is a good option, if there is any way around it. It defeats the point of using this type to ensure the value is actually serializable, because unknown could be a function.

Could `AlertTypeParams` type be something like `Record<string, SerializableState>`?

Reply:
it can just be `SerializableState` then, the Record part will be covered. I agree with stacey, enforcing this to actually be serializable makes sense

Reply:
We have been (perhaps too much 😬 ) lenient with rule type producers in how they define their `AlertTypeParams` (which is what would be considered the persistable state so when I change

```
type AlertTypeParams = Record<string, unknown>
```

to

```
type AlertTypeParams = SerializableState
```

I run into a very large number of typings issues from the various rule types, especially some of the plugins that use `io-ts` for runtime typings. Instead of trying to address them all in this issue, perhaps we can take a phased approach, to first get all the rule types to be compatible to `SerializableState` first?

Reply:
Is there a way to phase it so that the typescript leniency is isolated to the alerting plugin instead of changing the `SerializableValue` definition?  Once it's more lenient here, any downstream consumer could start misusing it, which will make it more difficult to undo later.